File: server/src/main/java/com/example/backend/modules/quiz/models/Answer.java
package com.example.backend.modules.quiz.models;

import com.example.backend.modules.history.models.HistoryAnswer;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Builder
@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table(name = "answers")
public class Answer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(length = 150,nullable = false)
    private String title;

    @Column(name = "is_correct",nullable = false)
    private Boolean isCorrect;

    @Column(name = "created_at",nullable = false)
    private Date createdAt;

    @Column(name = "updated_at",nullable = false)
    private Date updatedAt;

    @ManyToOne
    @JoinColumn(name = "question_id",nullable = false)
    @JsonBackReference
    private Question question;


    @OneToMany(fetch = FetchType.LAZY,cascade = CascadeType.ALL,mappedBy = "answer")
    @JsonManagedReference
    private List<HistoryAnswer> historyAnswers;
}

File: client/app/src/main/java/com/example/client/model/Answer.kt
package com.example.client.model

data class CreateAnswer(
    val title:String,
    val isCorrect:Boolean
)

data class Answer(
    val id:Int,
    val title:String,
    val correct:Boolean
){
    constructor(): this(
        -1,
        "",
        false
    )
}

data class EditAnswer(
    val id: Int,
    val title:String,
    val isCorrect: Boolean
)

data class EditListAnswer(
    val answers: List<EditAnswer>,
    val quizId: Int
)